numpy/lib/arraypad.py

```diff
@@ -74,6 +74,25 @@ def _round_ifneeded(arr, dtype):
         arr.round(out=arr)
 
 
+def _slice_at_axis(shape, sl, axis):
+    """
+    Construct a slice tuple the length of shape, with sl at the specified axis
+    """
+    slice_tup = (slice(None),)
+    return slice_tup * axis + (sl,) + slice_tup * (len(shape) - axis - 1)
```

Comment:
Yes, the last half is redundant - but it produces the same slices as we did before, and I see little reason to change that.